django/db/models/base.py

```diff
@@ -64,8 +79,9 @@ def __setstate__(self, args):
 
         class_dict['__reduce__'] = __reduce__
         class_dict['__setstate__'] = __setstate__
+        class_dict['model'] = attached_to
```

Comment:
Not sure if it's safe to do that as it creates a circular reference between `model` and the class. What about setting `__qualname__` directly instead by generating it from `'%s.%s' % (attached_to.__qualname__, name)`.

Reply:
Doing this would require the `pickle_exc_meta` function to then do an import to get the actual model object, since it's needed for the `getattr` it returns so the pickle can be unpickled. Not sure if that is preferable?

Reply:
@rmtobin are you referring to the unpickling logic of `ModelBase`? Are `models.Model` subclasses even pickleable if they haven't been discovered/loaded yet?

What I'm saying is that doing the following and removing the `copyreg.pickle` stuff should just work?

```python
if attached_to:
    class_dict['__qualname__'] = '%s.%s' % (attached_to.__qualname__, name)
```

Reply:
Oh I see what you mean. You are totally correct, it appears that works. This implementation was intended to work with Python 2, so this wasn't an option :) Updating.